File: ScottsExamples/TestStuff/BlankActivity36NavGraph/app/src/main/java/com/example/blankactivity36navgraph/SecondFragment.java
package com.example.blankactivity36navgraph;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

public class SecondFragment extends Fragment {
    TextView txt2;
    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_second, container, false);
    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        view.findViewById(R.id.button_second).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                NavHostFragment.findNavController(SecondFragment.this)
                        .navigate(R.id.action_SecondFragment_to_FirstFragment);
            }
        });

        txt2 = getView().findViewById(R.id.textview_second);
    }

    @Override
    public void onStart(){
        super.onStart();
        SecondFragmentArgs args = SecondFragmentArgs.fromBundle(getArguments());
        String message = args.getMessage() + " <-- this message was sent from  the first fragment";
        txt2.setText(message);


    }
}

File: chapter8/navigationdemo/app/src/main/java/com/example/navigationdemo/SecondFragment.kt
package com.example.navigationdemo

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.example.navigationdemo.databinding.FragmentSecondBinding


class SecondFragment : Fragment() {
    private var _binding: FragmentSecondBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentSecondBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.secondFragBtn.setOnClickListener {
            val action: SecondFragmentDirections.SecondFragmentToThirdFragment = SecondFragmentDirections.secondFragmentToThirdFragment()
            action.image = R.drawable.bot
            action.message2 = binding.secondFragTextview.text.toString()
            Navigation.findNavController(it).navigate(action)

        }
    }

    override fun onStart(){
        super.onStart()
        arguments?.let {
            val args = SecondFragmentArgs.fromBundle(it)
            binding.secondFragTextview.text = args.message1
        }
    }

}
